hazelcast/src/main/java/com/hazelcast/cache/ICache.java

```diff
@@ -944,4 +944,12 @@
      */
     CacheStatistics getLocalCacheStatistics();
 
+    /**
+     * Opens cache if available (not destroyed).
+     * If cache is already open, does nothing.
+     *
+     * @throws java.lang.IllegalStateException if cache is already destroyed
+     */
+    void open();
```

Comment:
Another way, is extending `ICacheInternal` from `ICache` and `AbstractHazelcastCacheManager` interact with `ICacheInternal` but gives `ICache` to API user since in this case `ICacheInternal` is a kind of `ICache`